File: AtCoder/ACP/BootCamp4b/Medium100_2/python/C_Colorful_Leaderboard.py
N = int(input())
A = list(map(int, input().split()))

used = [False] * 8
cnt = 0
for a in A:
    if a >= 3200:
        cnt += 1
    else:
        used[a // 400] = True
min_res = len(list(filter(lambda x: x, used)))
max_res = min_res + cnt
min_res = max(min_res, 1)
print(min_res, max_res)

File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/C_Colorful_Leaderboard.cpp
#include <bits/stdc++.h>
#define MAX_N 100
using namespace std;

int N;
int A[MAX_N];
bool used[8];

void solve() {
  int cnt = 0;
  for (int i = 0; i < N; ++i) {
    if (A[i] >= 3200) {
      ++cnt;
    } else {
      used[A[i] / 400] = true;
    }
  }
  int min_res = 0;
  int max_res = 0;
  for (int i = 0; i < 8; ++i) {
    min_res += used[i];
  }
  max_res = min_res + cnt;
  min_res += min_res == 0 && cnt;
  printf("%d %d\n", min_res, max_res);
}

int main() {
  cin >> N;
  for (int i = 0; i < N; ++i) {
    cin >> A[i];
  }
  solve();
}